fabric-screen-api-v1/src/main/java/net/fabricmc/fabric/impl/client/screen/ScreenEventFactory.java

```diff
@@ -33,190 +39,404 @@
 @ApiStatus.Internal
 @Environment(EnvType.CLIENT)
 public final class ScreenEventFactory {
+	// These lists will always have keep holding events, preventing them from being GC'd! This is bad!
+	// FIXME - Find a solution for this
+	public static final Set<Screen> ACTIVE_SCRRENS = new ReferenceArraySet<>(1);
```

Comment:
Oh whoops, this is indeed outdated, there used to be a list per event, and that list contained a record class with shenanigans that weren't going to work in the end

I chose reference because I wanted to have a low-overhead for this (*cries in libzoomer's performance massacre caused by set iterations*); According to my tests, only 1 should be there for most of the time, however, there are cases where there can be two screens on the set (like TitleScreen and  RealmsNotificationsScreen), so yeah, it can't be replaced by a variable; Also, it's a set instead of a list because the same instance of RealmsNotificationsScreen kept flooding the list, and while this could (and should) be fixed on the Quilt Screen API's side, who knows what other mods are doing something like it? So yeah, I wanted to avoid a huge disaster

...wait a minute, i just realized, scrrens, 🤦‍♀️

Reply:
Hm, I've sticked with ReferenceArraySet since yeah, it didn't cause any problems, after all, I never really deal with the objects themselves anyway